explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
-[`Window`](https://html.spec.whatwg.org/#the-window-object). It contains the following methods:
+* Chat, email, and social apps could signal that new messages have arrived.
+* Any application that needs to signal that user action is required (e.g., in a turn-based game, when it is the player's turn).
+* As a permanent indicator of a page's status (e.g., on a build page, to show that the build has completed).
 
-```webidl
-dictionary BadgeOptions {
-  // The scope the badge applies to.
-  // If unspecified this is the origin of the current page.
-  USVString scope;
-}
+### Use with Notifications
 
-[Exposed=Window]
-interface Badge {
-  // Sets the badge for a scope to be |contents|.
-  void set(unsigned long long contents, optional BadgeOptions options);
+The Badge API goes hand-in-hand with Notifications, since both APIs provide a
+way to give the user updates when they aren't directly looking at the page.
+Arguably, Badge could be part of Notifications, but we view it as a separate,
+complementary API.
 
-  // Sets the badge for a scope to be the special value, 'flag'.
-  void set(optional BadgeOptions options);
+Advantages of using the badging API over notifications:
 
-  // Clears the badge for a scope.
-  void clear(optional BadgeOptions options);
-}
-```
+* Can be used for much higher frequency / lower priority events than notifications, because each new event does not disrupt the user.
+* There may be no need to request permission to use the badging API, since it is much less invasive than a notification.
 
-> Note: Should we have a separate overload for boolean flags now, as discussed in [Issue 19](https://github.com/WICG/badging/issues/19) and [Issue 42](https://github.com/WICG/badging/issues/42)?
+Typically, sites will want to use both APIs together: notifications for high-importance events such as new direct messages or incoming calls, and badges for all new messages including group chats not directly addressed to the user.
 
-> Note: This API can only be used from a foreground page. Use from a service worker is being considered for the future. [The FAQ](#Why-cant-this-be-used-in-the-background-from-the-ServiceWorker-see-28-and-5) has more details).
+## Badge scope
 
-### Examples
+This section explains the difference between badging a *set of URLs* versus
+badging a *document*.
 
-#### Basic Examples
-These examples provided limited context. They are intended to show how the API will look.
+Consider four increasingly narrow sets of pages that a Badge is applied to:
 
-##### Setting an integer badge (as in an email app):
-```js
-Badge.set(getUnreadCount());
-```
+1. An entire origin (e.g., when you just want to set an unread count for the
+   entire site).
+2. All pages whose URL [starts
+   with](https://www.w3.org/TR/appmanifest/#dfn-within-scope) a given path
+   (e.g., if a site is organised into "projects", each with their own status
+   indicator).
+3. All pages with a specific URL path (e.g., if each page has a separate status
```

Comment:
I was thinking about this. I don't think scope is quite the solution we think it is.

It won't work for 
- Query strings (already discussed)
- SPAs on a fragment url (similar to query strings)
- Apps that serve pages on `/`, `/app`, `/app/page` and `/app/page/1` (there's no way to badge `/app` and only `/app`).

Reply:
I don't think any of these really change our calculation here. I think it's quite unlikely that people are going to want to badge arbitrary groupings of pages. That's why it's broken into these four categories. Almost always, you'll want to badge an entire site/app (which may be at a sub-path if your app shares an origin with other apps, but will _always_ have its own path).

Sometimes you might want to badge individual pages, which will work fine if you designed your URLs correctly. If not, you'll have to use the dynamic badge API.

This solves all the problems I care about.

Reply:
I kind of wish we didn't have to care about app scope either. Perhaps we could get away with not only supporting origin badges (if you have multiple apps on the same origin, tough, use the dynamic badging API).

It'd solve 90% of the use cases and be waaaay simpler. 

But probably not that big a deal.